src/pages/projects/Projects.tsx

```diff
@@ -0,0 +1,27 @@
+import { Outlet, useRouterState } from "@tanstack/react-router";
+import { generateBreadcrumbs } from "@/components/header/breadcrumb/generateBreadcrumbs";
+import Header from "@/components/header/Header";
+import { useState } from "react";
+import { useAuth } from "@/hooks/useAuth";
+
+const Projects: React.FC = () => {
+  const router = useRouterState();
+  const { username } = useAuth();
+
+  const currentPath: string = router.location.pathname;
+  const [breadcrumbs] = useState(generateBreadcrumbs(currentPath));
```

Comment:
This should not be a useState. Make it a `useMemo`

Reply:
Actually useState might also fit here, but it's weird bvecause you never setIt, and if someone will use this code he might change that and cause a lot of rerenders as it's an object